examples-builder/build.rs

```diff
@@ -23,21 +23,26 @@ fn build_elfs() {
     let _ = remove_file(&dest_path);
     let mut dest = File::create(&dest_path).expect("failed to create vars.rs");
 
-    // TODO(Matthias): skip building the elfs if we are in clippy or check mode.
-    // See git history for an attempt to do this.
+    let is_release = std::env::var("PROFILE").unwrap() == "release";
+    let mut args = vec!["build", "--examples", "--target-dir", "target"];
+    if is_release {
```

Comment:
This also resolved asked question in original issue https://github.com/scroll-tech/ceno/issues/632#issuecomment-2745315257